doc/HORDES.md

```diff
@@ -0,0 +1,62 @@
+existing hordes live in overmap::zg;
+hordes move with calls to overmap::move_hordes()
+These should both be removed, specific monster instances should be created instead of hordes, probably.
+
+monsters outside the reality bubble live in overmap::monster_map
```

Comment:
Yes, this whole file is simply my brain dump about the design of the feature, it will be cleaned up and corrected as I go, and tidied up into javadoc by the end of the process.